### PR TITLE
Add com.valvesoftware.steam.utility.gamescope as an extension of Steam

### PR DESCRIPTION
**Pull request by xinyazhang**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/Plagman/gamescope/issues/183**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub,] SourceForge).
    * This is a flatpak version of someone else's software, and the copyright holder owns the domain used in the application ID.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.valvesoftware.Steam.Utility.gamescope.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+appstream-compose: false
+build-options:
+  prefix: /app/utils/gamescope
+  prepend-path: /app/utils/gamescope/bin
+  append-pkg-config-path: /app/utils/gamescope/lib/pkgconfig
+  strip: true
+modules:
+  - modules/libinput/libinput.json
+  - modules/xwayland/xwayland.yml
+
```
Comment (barthalion) [inline com.valvesoftware.Steam.Utility.gamescope.yml:16]: Any reason these are split to separate files and one is in JSON?

Comment (xinyazhang) [reply com.valvesoftware.Steam.Utility.gamescope.yml:16]: The `libinput.json` is mostly copied from [tv.kodi.Kodi](https://github.com/flathub/tv.kodi.Kodi/blob/master/modules/libinput/libinput.json), while the `xwayland.yml` is written from scratch.

Comment (barthalion) [reply com.valvesoftware.Steam.Utility.gamescope.yml:16]: Ideally these should be merged into single file but this can be done after the merge.

**Comment by gasinvein** [comment]
I wonder how this is going to work. By launching the flatpak from a console session?

**Comment by xinyazhang** [comment]
> I wonder how this is going to work. By launching the flatpak from a console session?

Change the launching options in Steam. The details can be found at gamescope's README.

For example if you want to force 2X FXAA on an 1080p game, just use `gamescope -w 3840 -h 2160 -W 1920 -H 1080 -f -- %command%`

**Pull request closed without merging**